### PR TITLE
deploy_puppet - setup pollSCM and move logrotate into groovy

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

**Comment by ehelms** [comment]
Just asking since you are making changes here, should we update this to the naming scheme? e.g. puppet-deploy ?

**Comment by komidore64** [comment]
Ah yeah, that's probably a good call.

**Comment by komidore64** [comment]
@ehelms updated

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+- job:
+    name: puppet_deploy
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet_deploy.yaml]: Little thing, `-` instead of `_`

Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet_deploy.yaml]: updated!

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/puppetDeploy.groovy [hunk 2]
```diff
@@ -2,9 +2,14 @@ pipeline {
...
-        timestamps()
-        disableConcurrentBuilds()
         ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '3'))
+        disableConcurrentBuilds()
+        timestamps()
+    }
+
+    triggers {
+        pollSCM('H/5 * * * *')
```
Comment (akofink) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/puppetDeploy.groovy]: This polls the foreman-infra repo every 10 minutes?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+- job:
+    name: puppet-deploy
+    project-type: workflow
+    properties:
+      - github:
+          url: https://github.com/theforeman/foreman-infra
+    triggers:
+      - github
```
Comment (akofink) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet-deploy.yaml]: But is also triggered by git hooks from GitHub? Why do we need both?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/puppetDeploy.groovy (hunk 2 above)
Comment (ekohl) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/puppetDeploy.groovy]: every 5 I'd think

**Comment by ekohl** [comment]
When deploying we should be careful to remove the old job as well.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet-deploy.yaml (hunk 3 above)
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/puppet-deploy.yaml]: See https://wiki.jenkins.io/display/JENKINS/GitHub+Plugin -- the two are inter-related

**Comment by ehelms** [comment]
Thanks @komidore64 . Merging, I will delete old job once new job appears.

**Pull request merged**